dyn_em/module_initialize_real.F

```diff
@@ -3089,6 +3104,16 @@ SUBROUTINE init_domain_rk ( grid &
 
       !  Split NUDAPT  Urban Parameters
 
+      distributed_aerodynamics_if: IF (config_flags%use_distributed_aerodynamics) THEN
+         DO j = jts , MIN(jde-1,jte)
+            DO i = its , MIN(ide-1,ite)
+              IF (grid%ivgtyp(i, j) == model_config_rec%isurban(grid%id)) THEN
+                grid%frc_urb2d(i, j) = MAX(0.1, MIN(0.9, 1 - grid%shdavg(i, j) / 100.))
```

Comment:
not sure if this is a good way to estimate urban fraction (=1-vegfrac), but may be OK for now.

Reply:
OK, this makes sense, at least for now. more tests with the impact of shdavg vs shdmax are needed in the future.

Reply:
@tslin2 @epn09 OK, let me comment here since this is probably the first place where it showed up. I think I understood what Khanh is trying to do here. If there is no other data source for FRC_URB2D, one can come up with a few ways to assign FRC_URB2D values. It used to be that FRC_URB2D is prescribed in the URBPARM.TBL (e.g., 0.5 for low intensity residential urban etc). With NLCD data, FRC_URB2D is actually estimated from NLCD and is no longer using prescribed values from URBPARM.TBL. Now this part of code is trying to assign FRC_URB2D based on vegetation fraction, which is not a terrible idea. But this will limit the compatibility of this code with potentially new land use/land cover data that can provide FRC_URB2D. 

Khanh (first of all, congratulations on becoming a doctor, way to go), regarding ``If a WRF grid will all MODIS grids classified as urban, the WRF grid will have LANDUSEF(urban) = 100``, this is not true actually.  LANDUSEF is always computed based on land use land cover data in WPS, and stored and passed to WRF. But it is not used when the mosaic approach is not enacted. In a somewhat similar way, FRC_URB2D can be also computed using land use land cover data (especially with high resolution data like NLCD) and it is actually used by WRF (at least before LCZ was introduced, I don't know the details after LCZ is introduced).

Reply:
@DanLi-BU Thanks for your note about NLCD. Well-noted.